src/es/managers/PlacementManager.js

```diff
@@ -157,19 +157,26 @@ function positionCallout(callout, placementStrategy) {
 
   let isTargetFixed = isFixedElement(targetEl);
   let targetElBox = targetEl.getBoundingClientRect();
-  let calloutElBox = { width: callout.el.offsetWidth, height: callout.el.offsetHeight };
+  let calloutElBox = callout.el.getBoundingClientRect();
+  let arrowEl = callout.el.querySelector('.hopscotch-arrow');
+  let arrowElBox = arrowEl.getBoundingClientRect();
   let calloutPosition = placementStrategy.calculateCalloutPosition(
     targetElBox,
     calloutElBox,
-    callout.config.get('isRtl'),
-    callout.config.get('arrowWidth')
+    arrowElBox,
+    callout.config.get('isRtl')
     );
   
   //Adjust position if xOffset and yOffset are specified
   //horizontal offset
+  let placement = callout.config.get('placement');
   let xOffset = callout.config.get('xOffset');
   if (xOffset === 'center') {
-    calloutPosition.left = (targetElBox.left + targetEl.offsetWidth / 2) - (calloutElBox.width / 2);
+    if (placement === 'left' || placement === 'right') {
+      throw new Error('Can not use xOffset \'center\' with placement \'left\' or \'right\'. Callout will overlay the target.');
```

Comment:
I was going for an obvious error, so that person developing a tour will see and correct it. But I like your approach more. Developer will still see the error, but will have functioning tour at the same time.
I'll update the code.